vm/src/vm/vm_memory/memory.rs

```diff
@@ -158,6 +158,11 @@ impl AddressSet {
 
 pub struct Memory {
     pub(crate) data: Vec<Vec<MemoryCell>>,
+    /// Temporary segments are used when its necessary to write data, but we
+    /// don't know yet where it will be located. These segments are eventually
+    /// relocated to the main memory acording to `relocation_rules`. For
```

Comment:
```suggestion
    /// don't know yet where it will be located. These segments will eventually
    /// be relocated to the main memory according to the `relocation_rules`. For
```